core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java

```diff
@@ -600,7 +600,7 @@ private void connectBlock(final Block block, StoredBlock storedPrev, boolean exp
                         block.getHashAsString(), filteredTxHashList.size(), filteredTxn.size());
                 for (Sha256Hash hash : filteredTxHashList) log.debug("  matched tx {}", hash);
             }
-            if (expensiveChecks && block.time().getEpochSecond() <= getMedianTimestampOfRecentBlocks(head, blockStore))
+            if (expensiveChecks && block.time().compareTo(getMedianTimestampOfRecentBlocks(head, blockStore)) <= 0)
```

Comment:
We should be using `.isBefore()` and `.isAfter()` in code like this, rather than `.compareTo()`.  I probably should have noticed/commented earlier, but late is better than never.

Reply:
Are you ok with `!isAfter()` then? Or would you craft an `isBefore() || equals()`? That's why I went with `compareTo()`.

Reply:
Pushed a commit on top that uses `!isAfter()`.

Reply:
I think `!isAfter()` is pretty readable. In my mind I read it as "is not after" and it makes sense.